gatsby-node.js

```diff
@@ -1,20 +1,36 @@
+let fetch = require('node-fetch');
+
+const getRepository = async (slug) => {
+  const response = await fetch(`https://api.github.com/repos/mirego/${slug}`, {
+    headers: {
+      Authorization: `token ${process.env.GITHUB_TOKEN}`,
+    },
+  });
+
+  return await response.json();
+};
+
 exports.sourceNodes = async ({actions, createNodeId, createContentDigest}) => {
   const projects = require('./data/projects');
 
-  projects.forEach((project) => {
+  for (let project of projects) {
```

Comment:
Now that we need to call an async function (`getRepository`), we cannot use `forEach` anymore since we need to block the execution.

I haven’t played around with `async`/`await` much so I might write a quick TIL post about that!

Reply:
Waiting for each call that way will probably slow down the build. Maybe we could do something like this to fetch the data and create nodes in parallel:

```js
const projects = require('./data/projects');

const projectsPromises = projects.map(project => {
  return getRepository(project.slug).then(repository => {
    actions.createNode({
      ...project,
      id: createNodeId(project.name),
      starCount: repository.stargazers_count,
      createdAt: repository.created_at,
      internal: {
        type: 'Project',
        contentDigest: createContentDigest(project.name),
      },
    });
  });
});

Promise.all(projectsPromises);

Reply:
Oh! Awesome! 😍